torchvision/transforms/v2/functional/_utils.py

```diff
@@ -67,20 +74,25 @@ def register_kernel(dispatcher, datapoint_cls):
             f"but got {dispatcher}."
         )
 
-    if not (
-        isinstance(datapoint_cls, type)
-        and issubclass(datapoint_cls, datapoints.Datapoint)
-        and datapoint_cls is not datapoints.Datapoint
-    ):
+    if not (isinstance(datapoint_cls, type) and issubclass(datapoint_cls, datapoints.Datapoint)):
         raise ValueError(
             f"Kernels can only be registered for subclasses of torchvision.datapoints.Datapoint, "
             f"but got {datapoint_cls}."
         )
 
+    if datapoint_cls in _BUILTIN_DATAPOINT_TYPES:
+        raise ValueError(f"Kernels cannot be registered for the builtin datapoint classes, but got {datapoint_cls}")
```

Comment:
I don't think this is really needed. The previous error message was fine, but anyway.

Note that this means users can't override e.g. `erase` for bboxes, but this could be a desirable feature.

Reply:
Isn't that exactly what we wanted? We don't want users to register custom kernels for our builtin datapoints. Otherwise

https://github.com/pytorch/vision/blob/8faa1b14d383129877c7d233e8be848330980875/torchvision/transforms/v2/functional/_utils.py#L130-L134

would be valid again.

We could special case `F.erase` and the other outliers, but I feel users could come to us if they see the warning. We could need the input, since we ourselves are not 100% sure what we should actually do there.

Reply:
It's just that `erase` is a bit special in that we probably *could* have a kernel for it, we just haven't implemented it yet. So it's reasonable for users to want to have their own. But that's not super important.